lua/network_promises/http.lua

```diff
@@ -0,0 +1,108 @@
+NP.http = {}
+
+-- Returns a promise that resolves after t seconds
+-- fail : Should the promise reject after timeout
+function NP.timeout( t, fail )
+    local d = promise.new()
+    local method = fail and d.reject or d.resolve
+    timer.Simple( t, function()
+        method( d, "Timeout" )
+    end )
+    return d
+end
+
+local function responseSuccess( d, body, status, headers )
+    if math.floor( status / 100 ) == 2 then
+        d:resolve( body, status, headers )
+    else
+        d:reject( body, status, headers )
+    end
```

Comment:
```suggestion
    -- Is a 200-level response code
    local isSuccessStatus = math.floor( status / 100 ) == 2
    
    if isSuccessStatus then
        return d:resolve( body, status, headers )
    end

    d:reject( body, status, headers )
```

Thoughts?

Reply:
Depends on what promise reject returns, if it's nil then sure, else it could give unwanted behaviour